### PR TITLE
Add/update doc comments

### PR DESCRIPTION
**Pull request by HaoK**
@blowdart can you review these dev terse comments :)

Fixes https://github.com/aspnet/Security/issues/664

### CODE REVIEW

File: src/Microsoft.AspNetCore.Authentication/IClaimsTransformer.cs [hunk 1]
```diff
@@ -6,8 +6,16 @@
...
 {
+    /// <summary>
+    /// Used for claims transformation.
+    /// </summary>
     public interface IClaimsTransformer
     {
+        /// <summary>
+        /// Transform a ClaimsPrincipal.
+        /// </summary>
+        /// <param name="principal"></param>
```
Comment (PinpointTownes) [inline src/Microsoft.AspNetCore.Authentication/IClaimsTransformer.cs]: Empty docs? :smile:

File: src/Microsoft.AspNetCore.Authorization/AuthorizationOptions.cs [hunk 2]
```diff
@@ -47,6 +60,11 @@ public void AddPolicy(string name, Action<AuthorizationPolicyBuilder> configureP
             PolicyMap[name] = policyBuilder.Build();
         }
 
+        /// <summary>
+        /// Return the policy for the given name, or null.
+        /// </summary>
+        /// <param name="name"></param>
```
Comment (PinpointTownes) [inline src/Microsoft.AspNetCore.Authorization/AuthorizationOptions.cs]: Empty docs? :smile:

Comment (HaoK) [inline src/Microsoft.AspNetCore.Authorization/AuthorizationOptions.cs]: Is that against the rules?  Name is pretty self documenting!

File: src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectOptions.cs [hunk 3]
```diff
@@ -17,7 +17,7 @@
 namespace Microsoft.AspNetCore.Builder
 {
     /// <summary>
-    /// Configuration options for <see cref="OpenIdConnectOptions"/>
+    /// Configuration options for <see cref="OpenIdConnectMiddleware"/>
```
Comment (blowdart) [inline src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectOptions.cs]: Consistency - either use the style from CookieAuthenticationOptions, or change it to match this. Personally I like Contains the options used by X.

File: src/Microsoft.AspNetCore.Authentication/IClaimsTransformer.cs (hunk 1 above)
Comment (blowdart) [inline src/Microsoft.AspNetCore.Authentication/IClaimsTransformer.cs]: Lazy :p 

Summary : Provides a central transformation point to change the specified principal.
Param: The principal to transform.
Returns: The transformed principal.

File: src/Microsoft.AspNetCore.Authorization/AuthorizationOptions.cs (hunk 2 above)
Comment (blowdart) [inline src/Microsoft.AspNetCore.Authorization/AuthorizationOptions.cs]: You still need to expand.

/// <summary>
/// Returns the policy for the specified name, or null if a policy with the name does not exist.
/// </summary>
/// <param name="name">The name of the policy to return.</param>
/// <returns>The policy for the specified name, or null if a policy with the name does not exist.</returns>

File: src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectOptions.cs (hunk 3 above)
Comment (HaoK) [inline src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectOptions.cs]: Cookies is the outlier, fixed that to match the rest

**Comment by HaoK** [comment]
Updated adding all these words....

**Comment by HaoK** [comment]
b5300ad0e4c48a9d69b3be27df8bed9db3f959b2

**Pull request closed without merging**